$DragonFly: src/gnu/usr.bin/groff/src/preproc/eqn/patches/neqn.sh.no_obj.patch,v 1.1 2005/09/01 20:47:13 asmodai Exp $

--- neqn.sh	2005-08-31 14:28:07.000000000 +0200
+++ neqn.sh	2005-08-31 15:32:38.000000000 +0200
@@ -2,9 +2,23 @@
 # Provision of this shell script should not be taken to imply that use of
 # GNU eqn with groff -Tascii|-Tlatin1|-Tutf8|-Tcp1047 is supported.
 
+# Default device.
+case "${LC_ALL-${LC_CTYPE-${LANG}}}" in
+  *.UTF-8)
+    T=utf8 ;;
+  iso_8859_1 | *.ISO*8859-1 | *.ISO*8859-15)
+    T=latin1 ;;
+  *.IBM-1047)
+    T=cp1047 ;;
+  *.KOI8-R)
+    T=koi8-r ;;
+  *)
+    T=ascii ;;
+esac
+
 @GROFF_BIN_PATH_SETUP@
 PATH="$GROFF_RUNTIME$PATH"
 export PATH
-exec @g@eqn -Tascii ${1+"$@"}
+exec @g@eqn -T${T} ${1+"$@"}
 
 # eof
